src/material-experimental/mdc-slider/slider-thumb-decorator.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import {Thumb} from '@material/slider';
+
+/**
+ * Handles displaying the slider knobs and their value indicators.
+ */
+@Component({
+  selector: 'mat-slider-start-thumb-decorator, mat-slider-end-thumb-decorator',
+  templateUrl: 'slider-thumb-decorator.html',
+  host: {
+    'class': 'mdc-slider__thumb',
+    '(mouseenter)': '_mouseenter.emit()',
```

Comment:
Who would listen to these events? If they're only here for internal consumption, we could bind them manually from the parent that listens to them.

Reply:
The plan is to create a child component `MatSliderKnob` which handles all of the ripple states of the MDC slider. It would need to know when the `MatSliderThumbDecorator` is hovered. Is there a different/preferred way to listen for these events from a child component?

Reply:
You can always listen to events on an HTML element, regardless of whether its an Angular component or not. My gut feeling is that you can just roll the template for this component into the `<mat-slider>` template and add listeners for elements you care about in `MatSlider`.

Reply:
I wrote my reason for extracting this into a separate component above. If you decide we'd prefer this to be combined back into the `MatSlider` class, I'll make these changes.

Reply:
+1 to doing this in `MatSlider` itself